### DISCUSSION TITLE
3era Entrega - Consulta Tarjetas Colaboradores

### DISCUSSION THREAD

**Discussion by matiastito** [dds-jv/foro#435] [TPA - 2024] [answered]
Hola a todos!

Espero estén bien, los molestamos porque al leer el enunciado de la 3era entrega:

> Los colaboradores que deseen colocar viandas o colaborar en la distribución de las mismas contarán con una tarjeta (al igual que las personas con necesidad). La entrega de las tarjetas a los colaboradores se hará a través del correo a la dirección de entrega especificada en el momento que escogieron contribuir entregando viandas o distribuyendo las mismas. Para resolver esta tarea, se contará con una persona operadora que revisará periódicamente las solicitudes de tarjetas.

1. Cuando dice 'corre', se refiere a 'correo postal'
2. Nos nos queda claro cómo se 'entera' el Sistema que el correo entregó la tarjeta.
3. Qué es lo que revisa la 'persona operadora'.

Esto por ahora, ¡muchas gracias!

**Comment by flbulgarelli** [comment]
👋 ¡Hola!

> Cuando dice 'corre', se refiere a 'correo postal'

Sí 

> Nos nos queda claro cómo se 'entera' el Sistema que el correo entregó la tarjeta

A decir verdad, no está especificado. En muchos sistemas reales este último paso es totalmente manual, por mensaje de texto o aún por llamada telefónica. Dado que la gestión del envío es manual y queda en gran medida fuera del alcance del sistema (más allá de proveer la información para que una persona operadora del sistema la use), la confirmación de recepción también la manejaría de forma manual. De todas formas, ¿para que necesitan eso?


> Qué es lo que revisa la 'persona operadora'.

Se encarga de ver el listado de solicitudes de tarjetas pendientes. Con esa información, la persona se encargará de confeccionar planillas digitales o en papel con las que organizará la logística para enviar las tarjetas. Quizás sea ella misma quien haga esa tarea de repartirlas, o quizás se comunique por correo, telefónicamente o a través de otro sistema con quien lo haga. 

Es posible que está persona también se encargue de confeccionar los remitos, y/o que algunas de todas estas tareas las realice mentalmente o en medios aún más rudimentarios, como un cuaderno. 

En cualquier caso, queda fuera del alcance del sistema software que estamos construyendo: el mismo solo debe proveer la información necesaria para que está persona sepa quien está esperando por una tarjeta y dónde se encuentra. 

¡Espero que haya quedado más claro!

**Reply by matiastito** [reply]
Muchas gracias por la respuesta.

Nos quedó un poco mas claro; entiendo que el Sitema debe poder informar las Solicitudes de Tarjeta pendientes pero sigo teniedo dudas con respecto a:
- Quién es el encargado de setearle al Colaborador la tarjeta (¿la UI ofrece al Operador la posibilidad de tipear el Codigo de Tarjeta que va a despachar?).
- Si las Solicitudes tienen algun comportamiento para hacer el seteo de la tarjeta al colaborador para poder efectivizar la relación Colaborador<->Tarjeta. Algo así `new SolicitudDeTarjeta(Colaborador).cerrarSolicitud(Tarjeta)`.

Muchas gracias, perdón por molestar!

**Reply by flbulgarelli** [reply]
> * ¿la UI ofrece al Operador la posibilidad de tipear el Codigo de Tarjeta que va a despachar?

Sí. Se me ocurren varias sofisticaciones que se podrían hacer, desde cargas masivas hasta lectura automática del número de tarjeta usando un lector, pero todo eso está largamente fuera del alcance de la entrega. 

Por ahora, con que el sistema permita listar esas ordenes pendientes y crear la asociación entre una persona y su tarjeta, es suficiente.

**Reply by flbulgarelli** [reply, marked as answer]
> * Si las Solicitudes tienen algun comportamiento para hacer el seteo de la tarjeta al colaborador para poder efectivizar la relación Colaborador

En cuanto a esto, es una cuestión de diseño que queda de tu lado definir.

**Reply by matiastito** [reply]
Gracias!
